src/handler/recalls.ts

```diff
@@ -0,0 +1,90 @@
+import { SecretsManager } from '@dvsa/aws-utilities/classes/secrets-manager-client';
+import { getProfile } from '@dvsa/cvs-feature-flags/profiles/vtx';
+import { EnvironmentVariables } from '@dvsa/cvs-microservice-common/classes/misc/env-vars';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { MotSecret } from '../models/motRecalls';
+import { formatErrorMessage } from '../util/errorMessage';
+import { addHttpHeaders } from '../util/httpHeaders';
+import logger from '../util/logger';
+import {
+  filterMotRecalls, getBearerToken, getMotRecallsByVin,
+} from '../util/recalls';
+import { validateSingleVin } from '../validators/recalls';
+
+const cache: Map<string, (string | MotSecret)> = new Map();
+const defaultResponse = addHttpHeaders({
+  statusCode: 200,
+  body: JSON.stringify({
+    manufacturer: null,
+    hasRecall: false,
+  }),
+});
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Recalls end point called');
+  try {
+    const featureFlags = await getProfile();
+
+    if (!featureFlags.recallsApi) {
+      logger.error('Recall Feature Flag is undefined');
+      return addHttpHeaders({
+        statusCode: 500,
+        body: 'Recall Feature Flag is undefined',
+      });
+    }
+
+    if (!featureFlags.recallsApi.enabled) {
+      logger.warn('Flag disabled: please enable for recalls functionality');
+      return defaultResponse;
+    }
+
+    const vin: string = decodeURIComponent(event.pathParameters?.vin as string);
+    if (!validateSingleVin(vin)) {
+      logger.error(formatErrorMessage('VIN provided in path parameter is not valid.'));
+      return defaultResponse;
+    }
+
+    const cachedMotSecret = cache.get('motSecret') as MotSecret;
+    const motSecret = cachedMotSecret ?? await SecretsManager.get(
+      { SecretId: EnvironmentVariables.get('MOT_RECALL_SECRET') },
+      {},
+      { fromYaml: true },
+    );
+
+    if (!cache.has('motSecret')) {
+      cache.set('motSecret', motSecret);
+    }
+
+    const cachedBearerToken = cache.get('bearerToken');
+    const bearerToken = cachedBearerToken ?? await getBearerToken(motSecret);
+
+    if (!bearerToken) {
+      logger.error('bearer token not found');
+      return defaultResponse;
+    }
+
+    if (!cache.has('bearerToken')) {
+      cache.set('bearerToken', bearerToken);
+    }
+
+    const recalls = await getMotRecallsByVin(vin, cache, motSecret);
```

Comment:
I like what you have done with the caching above, where did we land on checking the expiryDate of the token prior to using it to make this getMotRecallsByVin call?

It is something we are going to want to put in at some point, wondering if it's worth adding the logic in for that now we have this wired up and working?

Reply:
Imo we leave it. 35 min tokens makes this a waste of time and something that will basically never happen.

Reply:
Okay, there is sufficient logging there for us to monitor this for now. If we do start seeing the error message from a 401/403's being returned when the FE start using this, then it's definitely something we will need to look at.